### PR TITLE
Support big/little/both endians on signature verification on SPDM1.0/1.1

### PR DESCRIPTION
**Pull request by richkong88**
Changes to support runtime decision on endianness (BE\LE\Both) for signature verification on SPDM1.0/1.1.

### CODE REVIEW

**Comment by jyao1** [comment]
@richkong88, thanks for the update.

I suggest to do the crypto handling (swap, double verification) in spdm_crypto_lib, to make the solution more scalable.
We will have more command requiring signing verification in the future 1.4.
I dont want to bring the burden to SPDM command writer. I suggest we HIDE the logic.

My comment for current API in spdm_crypto_lib, is to add a new API, dont change old API.
The old API implementation can invoke the new API, with default endianness.

**Comment by richkong88** [comment]
@jyao1 Ok. I can do it the way you suggest. I prefer that too. I just did not want to change any APIs so I did it this way.

File: include/internal/libspdm_common_lib.h [hunk 1]
```diff
@@ -565,6 +565,10 @@ typedef struct {
 #if LIBSPDM_FIPS_MODE
     libspdm_fips_selftest_context fips_selftest_context;
 #endif /* LIBSPDM_FIPS_MODE */
+
+    /* Endianness (BE/LE/Both) to use for signature verification on SPDM 1.0 and 1.1 */
```
Comment (jyao1) [inline include/internal/libspdm_common_lib.h:569]: Maybe add comment say: This field is ignored on SPDM 1.2 and above.

**Comment by richkong88** [comment]
Moving to different PR.

**Pull request closed without merging**